packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -36,7 +36,8 @@ export default util.createRule<Options, MessageIds>({
           parent &&
           (parent.type === AST_NODE_TYPES.VariableDeclarator ||
             parent.type === AST_NODE_TYPES.CallExpression ||
-            parent.type === AST_NODE_TYPES.TSAsExpression ||
+            (parent.type === AST_NODE_TYPES.TSTypeAssertion &&
+              parent.kind === 'as') ||
```

Comment:
i'm unsure about this one, as it look like a issue in current code base, old type assertion and as expression should be threated in same way here

Reply:
fixed in #3074